src/module/system/damage/formula.ts

```diff
@@ -107,7 +106,7 @@ function createDamageFormula(
 
     // Add modifiers
     for (const modifier of damage.modifiers.filter((m) => m.enabled && outcomeMatches(m))) {
-        const matchingBase = baseTerms.find((b) => b.damageType === modifier.damageType) ?? baseTerms[0];
+        const matchingBase = damage.base.find((b) => b.damageType === modifier.damageType) ?? damage.base[0];
```

Comment:
```suggestion
        const matchingBase = damage.base.find((b) => b.damageType === modifier.damageType) ?? damage.base.at(0);
        if (!matchingBase) continue;
```

Reply:
damage.base is explicitly checked to have a length at the start of the function, but typescript is too dumb to realize that.

If we even could though, I'm not sure if this is the correct behavior if we don't have a base at all. Perhaps it'd be better to fall back to untyped.